### ISSUE TITLE
when the state of alert increase from warn or crit to unknown, why not send unknown notification?

### ISSUE THREAD

**Issue by batmanwjw** [bosun-monitor/bosun#1812]
my config like this:

```
tsdbHost = 192.168.18.73:4242
stateFile = /usr/share/bosun/bosun.state
redisHost = localhost:6379
redisDb = 1
checkFrequency = 10s

template default_post {
    subject = `{"alert_key": "{{.AlertKey}}", "level": "{{.Last.Status}}", "worst_level": "{{.WorstStatus}}",
                "alert_rule": "{{.Alert.Name}}", "host": "{{.Group.host}}", "time": "{{.Last.Time}}"}`
}

template default_unknown {
    subject = `{"name": "{{.Name}}", "time": "{{.Time}}", "group": "{{.Group}}", "level": "unknown"}`
}

notification default_post {
    post = http://localhost:18018/v1/alerts/notify
    body = {{.}}
    contentType = application/json
    next = default_post
    timeout = 30s
    runOnActions = false
}

unknownTemplate = default_unknown

alert bf285adc-aa29-487c-9357-6d5c47812b44 {
    template = default_post
    runEvery = 2
    $metriccpu = q("avg:cpu.idle.percent{host=c9dd2b1c-bb19-4e5c-aa9d-5cd53e353eb9}", "30s", "")
    $avgcpu = avg($metriccpu)
    warn = $avgcpu > 10
    warnNotification = default_post
    crit = $avgcpu > 20
    critNotification = default_post
}# end alert bf285adc-aa29-487c-9357-6d5c47812b44
```

when the host triggers warn alert, I can receive the warn notification, then I stop send data to opentsdb. Then the state of alert increase to unknown, but I can't receive the unknown notification.Is this right?

**Comment by kylebrandt** [comment]
The behavior you *expect* is correct. Will need to try to figure out what it is going on here. Maybe @captncraig can at least take a quick look and see if anything comes to mind on why this might happen, after that I can dig deeper

**Comment by pradeepbbl** [comment]
We see something very similar to what is described here:

We are using graphite as our primary data source and sometimes we do get missing metrics or data points which resulted into empty response, prior to bosun-0.6.0 we used to get unknowns (if ignoreUnkown or unkonwIsNormal not set) in these cases but after moving to bosun-0.6.0 we stop getting unknowns and only seeing the error getting logged in bosun error page.

```
2017/10/17 11:04:05 error: check.go:571: Error checking alert balancer_config_drift_alert: graphite ParseError (http://XXXXXXXXXXXXXXXXXX/render/?format=json&from=1508230564&target=alias%28sum%28general.balancer.%2Alb%2A.haproxy.config.drift%29%2C+%27drift%27%29&until=1508230864): empty response
2017/10/17 11:04:05 info: check.go:579: check alert balancer_config_drift_alert done (345.942723ms): 0 crits, 0 warns, 0 unevaluated, 0 unknown
```

The above behavior causing two problems mentioned below 

* The error page in bosun UI getting slow over time
* It's very difficult to convince our users to monitor the bosun error page regularly or in more proactive manner

https://github.com/bookingcom/bosun/commit/3eb7ec39fc2e539bab5d5a31402305a4b53d5716

With this patch we are making use of the ignoreUnknown directive and able to trigger unknown if backend fails to fetch result e.g empty response 

We have also introduced a new alert directive ignoreError, when set to true the error will not log to an error page. This is useful when we are alerting on graphite metrics for a number of errors where sometimes the absence of recent data means that there are no errors and that the state should go back to normal. So in these cases, this option helps reduce spam in error page and load faster.
